rama-tcp/src/client/connect.rs

```diff
@@ -23,6 +23,49 @@ use tokio::{
     },
 };
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+enum IPModes {
+    Dual,
+    SingleIpV4,
+    SingleIpV6,
+    DualPreferIpV4
+}
+
+impl Default for IPModes {
+    fn default() -> Self {
+        Self::Dual
+    }
+}
+
+//DNS Resolver
+#[derive(Clone)]
+struct DnsResolveIpMode<D>{
+    resolver: D,
+    mode: IPModes
+}
+
+impl<D> DnsResolveIpMode<D>{
+    fn new(resolver:D, mode: IPModes) -> Self {
+        Self { resolver, mode}
+    }
+}
+
+struct ConnectIpMode<C>{
+    connector: C,
+    ip_mode: IPModes
+}
+
+impl<C>ConnectIpMode<C>{
+    fn new(connector: C, ip_mode: IPModes) -> Self {
+        Self {connector, ip_mode}
+    }
+}
+
+/*
+TODO - Using the wrapper types
+in the tcp_connect fn, update the dns type to use the DnsResolveIpMode
```

Comment:
I think it's in `async fn tcp_connect_inner<State, Dns, Connector>(` that you'll want to check for `DnsResolveIpMode`.

And it's in the location where we actually establish a tcp stream that you'll check for `ConnectIpMode`.

Originally I was thinking to also automatically let one influence the other, but I think it makes more sense to just let the user worry about that. E.g. if user selects Ipv6-only for DNS and ipv4-only for Connect, then that obviously will fail, but that is up to the user to then fix. No point in trying to do anything about that, as the only reasonable thing to do would be to return some kind of error, so let's not bother with that as the current error will already be fine enough.